### PR TITLE
feat(FieldLabel): Added concept of FieldLabel

### PR DESCRIPTION
**Pull request by dborysiewicz**
Added concept the of FieldLabel and switched all inputs (besides Checkbox) to utilize this, behind the scenes it uses Text (which is responsive at this time 14px -> 18px <740px) but also sets text to bold <740px. 

One downside was this is not usable in Checkbox however the same rules (14px -> 18px & bold for <740px) has been applied for consistency.

### CODE REVIEW

File: react/fields/Checkbox/Checkbox.less [hunk 1]
```diff
@@ -16,11 +16,15 @@
 }
 
 .standard {
-  .touchableText(@standard-type-scale);
+  .touchableText(@standard-type-scale-mobile);
```
Comment (michaeltaranto) [inline react/fields/Checkbox/Checkbox.less]: 🙌

File: react/fields/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -70,6 +50,7 @@ export default class Dropdown extends Component {
         return new Error(`\`inputProps.id\` will be overridden by \`id\` in ${componentName}. Please remove it.`);
       }
     },
+    /* eslint-enable consistent-return */
```
Comment (michaeltaranto) [inline react/fields/Dropdown/Dropdown.js]: Nice catch!

File: react/fields/Dropdown/Dropdown.js [hunk 3]
```diff
@@ -178,7 +137,7 @@ export default class Dropdown extends Component {
 
     return (
       <div className={classNames}>
-        {this.renderLabel()}
+        <FieldLabel {...this.props} />
```
Comment (michaeltaranto) [inline react/fields/Dropdown/Dropdown.js]: I think in the case of `FieldLabel` and `FieldMessage` we shouldnt be blindly spreading props, rather cherry picking what should flow through to them. Same probably goes for the input/select renderer, we should only be spreading the not-message and not-label related props. Thoughts?

Comment (dborysiewicz) [inline react/fields/Dropdown/Dropdown.js]: The idea behind this is that the prop validation (special rules around label, id & labelProps, etc) can live inside FieldLabel. Happy to look at alternatives if this is still possible?

File: react/fields/FieldLabel/FieldLabel.less [hunk 4]
```diff
@@ -0,0 +1,11 @@
+@import (reference) "~seek-style-guide/theme";
+
+.label {
+  @media only screen and (max-width: (@responsive-breakpoint - 1)) {
+    font-weight: @sk-bold;
+  }
+}
+
+.secondaryLabel {
+  font-weight: 300;
```
Comment (mengtzu) [inline react/fields/FieldLabel/FieldLabel.less]: Shouldn't this either be an existing weight, or we should be creating a new variable for it?

File: react/fields/Dropdown/Dropdown.js (hunk 3 above)
Comment (michaeltaranto) [inline react/fields/Dropdown/Dropdown.js]: Sure, the validation I understand living in the component that cares. I'm not sure about blindly passing everything though. @markdalgleish you have any thoughts on this?

File: react/fields/FieldLabel/FieldLabel.less [hunk 5]
```diff
@@ -0,0 +1,11 @@
+@import (reference) "~seek-style-guide/theme";
+
+.label {
+  @media only screen and (max-width: (@responsive-breakpoint - 1)) {
+    font-weight: @sk-bold;
+  }
+}
+
+.secondaryLabel {
+  font-weight: @sk-light;
```
Comment (michaeltaranto) [inline react/fields/FieldLabel/FieldLabel.less]: Has this been validated by the design team? Is a `secondaryLabel` the same weight regardless of screen size, even though the `label` is changing weight?

File: react/fields/TextField/TextField.less [hunk 6]
```diff
@@ -5,12 +5,6 @@
   position: relative;
 }
 
-.label {
-  .standardText();
-  display: inline-block;
-  margin-bottom: @grid-row-height;
-}
```
Comment (michaeltaranto) [inline react/fields/TextField/TextField.less]: 👋

File: react/fields/Dropdown/Dropdown.js (hunk 3 above)
Comment (markdalgleish) [inline react/fields/Dropdown/Dropdown.js]: I guess the middle ground here is we pluck the props that the `FieldLabel` cares about and it should still work, right?

Comment (markdalgleish) [inline react/fields/Dropdown/Dropdown.js]: Sorry, that was a little unclear, I mean like this:
```js
const { foo, bar, baz } = this.props;
<FieldLabel {…{ foo, bar, baz }} />
```

File: react/fields/FieldLabel/FieldLabel.less (hunk 5 above)
Comment (dborysiewicz) [inline react/fields/FieldLabel/FieldLabel.less]: secondaryLabel is really a internal/private implementation used for the max character wording in the TextArea however I will run it by Steph.

File: react/fields/Dropdown/Dropdown.js (hunk 3 above)
Comment (dborysiewicz) [inline react/fields/Dropdown/Dropdown.js]: If you explicitly grab the props required for FieldLabel you have to have the matching propType validation in every single component that is using FieldLabel :(

File: react/fields/FieldLabel/FieldLabel.less (hunk 5 above)
Comment (dborysiewicz) [inline react/fields/FieldLabel/FieldLabel.less]: @stephaniemoss when you get a chance can you let us know about this? In this context secondaryLabel is used in TextArea when `Max Character` options is passed in, example at: http://seek-style-guide--53438087bf8ed17f0a3d3cbaacc79035f261ebf2.surge.sh/textarea

Comment (stephaniemoss) [inline react/fields/FieldLabel/FieldLabel.less]: Let's stick with the default way it currently is now, I don't want to be kind of half tweaking this change on individual components. It really should be formed as patterns on a higher level.

Thanks!

File: react/fields/Dropdown/Dropdown.js [hunk 7]
```diff
@@ -135,12 +135,15 @@ export default class Dropdown extends Component {
       [className]: className
     });
 
+    // eslint-disable-next-line react/prop-types
+    const { id, label, labelProps, secondaryLabel, invalid, help, helpProps, message, messageProps } = this.props;
+
     return (
       <div className={classNames}>
-        <FieldLabel {...this.props} />
+        <FieldLabel id={id} label={label} labelProps={labelProps} secondaryLabel={secondaryLabel} />
```
Comment (markdalgleish) [inline react/fields/Dropdown/Dropdown.js]: I'd write this as `<FieldLabel {...{ id, label, labelProps, secondaryLabel }} />`

File: theme/type/type.less [hunk 8]
```diff
@@ -44,5 +44,6 @@
 @interaction-type-scale: 1.8;
 
 // Text variants
+@sk-light: 300;
```
Comment (markdalgleish) [inline theme/type/type.less]: Should we be documenting this?

File: react/fields/FieldLabel/FieldLabel.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    const { labelProps, id } = this.props;
+    const allLabelProps = {
+      ...combineClassNames(labelProps, styles.label),
+      ...(id ? { htmlFor: id } : {})
+    };
+    return (
+      <label {...allLabelProps}>
+        <Text>{label} {this.renderSecondary()}</Text>
```
Comment (markdalgleish) [inline react/fields/FieldLabel/FieldLabel.js]: 🙌

**Comment by markdalgleish** [comment]
Seriously awesome work consolidating all the labels into a single component. As discussed, even if we iterate on the design, the amount of work to do so will now be trivial 👏 

🎨 📦 🚀

**Pull request merged**